frontend-ssr/pages/submissions.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Head from 'next/head';
+import Footer from '~/components/footer/Footer.component';
+import Header from '~/components/header/Header.component';
+
+import styles from '../styles/pages/default-page.module.scss';
+import style from './submissions.module.scss';
+import ExSubmission from '../components/exercise-submissions/ExerciseSubmission';
+import ExerciseService from '../services/ExerciseSubmissions.service';
+import { loadSubmissions } from '~/redux/exercise-submissions/exercise-submissions.actions';
+import SubmissionSkeleton from '~/components/exercise-submissions/ExerciseSubmissionSkeleton';
+import SweetAlertService from '../services/sweet-alert/SweetAlert.service';
+import store from '~/redux/store';
+
+interface State {
+  loading: boolean;
+}
+interface Props {
+  submissions: any;
+  dispatch: any;
+}
+
+class Submissions extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
+  componentDidMount() {
+    let { submissions } = this.props;
+    const state = submissions;
+    function shouldFetchSubmissions() {
+      if (state.submissions === undefined) {
+        return true;
+      }
+      return false;
+    }
+    if (shouldFetchSubmissions()) {
+      this.getData();
+    } else {
+      store.getState();
+      this.setState({
+        loading: false,
+      });
+    }
+  }
+
+  getData = async () => {
+    const { dispatch } = this.props;
+    try {
+      await ExerciseService.getSubmissions(0, '')
```

Comment:
You either use `async-await` or `Promises and .then`. With a few exceptions we usually don't combine the both, which you do here.

This leads me to believe you haven't quite understood `async-await` yet. :D Give it another read and then look over this code again.